### ISSUE TITLE
Automatic derivation of a configured encoder/decoder fails for multiple fields including a newtype in a case class

### ISSUE THREAD

**Issue by tarao** [circe/circe#688]
This is an update from #641.

`io.circe.generic.extras.AutoDerivation` fails to find an encoder/decoder for a case class with multiple fields of `shapeless.newtype.Newtype[]`.

If there is only one field in the case class, it compiles.  If there are multiple fields (not necessarily newtypes but at least one newtype) in a case class, it fails.

```scala
  import io.circe.{Decoder, Encoder}
  import io.circe.generic.extras.auto._
  import io.circe.generic.extras.defaults._
  import shapeless.newtype
  import shapeless.syntax.unwrapped._

  type URL = newtype.Newtype[String, URL.Tag]
  object URL {
    trait Tag

    def apply(url: String): URL = newtype(url)
  }

  // case class Item(url: URL) // this compiles
  case class Item(url1: URL, url2: URL) // this fails
  // case class Item(url1: URL, url2: String) // this also fails

  implicit val encoder: Encoder[URL] =
    Encoder[String].contramap(_.unwrap)

  implicit val decoder: Decoder[URL] =
    Decoder[String].map(URL(_))

  Encoder[Item]
```

**Comment by travisbrown** [comment]
Thanks for the test case, will take a look as soon as possible!

**Comment by tarao** [comment]
It seems have been fixed since v0.9.0.

My colleague @tanishiking investigated a bit and he said it looks a bug in Shapeless, which have been fixed since Shapeless 2.3.3:
https://github.com/milessabin/shapeless/pull/761
